pulsar-common/src/main/java/org/apache/pulsar/common/protocol/PulsarDecoder.java

```diff
@@ -126,17 +126,12 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             }
             messageReceived();
 
+            safeInterceptCommand(cmd);
+
             switch (cmd.getType()) {
             case PARTITIONED_METADATA:
                 checkArgument(cmd.hasPartitionMetadata());
-                try {
-                    interceptCommand(cmd);
-                    handlePartitionMetadataRequest(cmd.getPartitionMetadata());
-                } catch (InterceptException e) {
-                    writeAndFlush(ctx,
-                            Commands.newPartitionMetadataResponse(getServerError(e.getErrorCode()),
-                            e.getMessage(), cmd.getPartitionMetadata().getRequestId()));
-                }
+                handlePartitionMetadataRequest(cmd.getPartitionMetadata());
```

Comment:
This is a behavior change if `InterceptException` is thrown from `interceptCommand`. This may cause client stuck if it can't get the error.

Reply:
> The `safeInterceptCommand()` method catches the `Exception`, it is safe.

Yes, it catch the exception, but it seems there is no `writeAndFlush` in `safeInterceptCommand`. The behavior is different to clients.